wrappers/src/results_cs.cpp

```diff
@@ -124,20 +123,16 @@ REALM_EXPORT Results* results_get_filtered_results(const Results& results, uint1
 {
     return handle_errors(ex, [&]() {
         Utf16StringAccessor query_string(query_buf, query_len);
-        auto query = results.get_query();
         auto const &realm = results.get_realm();
 
-        parser::ParserResult result = parser::parse(query_string.to_string());
-
-        parser::KeyPathMapping mapping;
+        query_parser::KeyPathMapping mapping;
         realm::populate_keypath_mapping(mapping, *realm);
 
-        query_builder::NoArguments no_args;
-        query_builder::apply_predicate(query, result.predicate, no_args, mapping);
+        query_parser::NoArguments no_args;
+        auto query = results.get_table()->query(query_string, no_args, mapping);
+        auto ordering = query.get_ordering() ? *query.get_ordering() : results.get_descriptor_ordering();
```

Comment:
query.get_ordering() returns the ordering, but it also removes it from the query, so you can't call it twice (this is a bit of an unusual pattern). Try this: 
```
    query_parser::KeyPathMapping mapping;
    realm::populate_keypath_mapping(mapping, *this->get_realm());
    Query parsed_query = get_query().get_table()->query(query_string, no_args, mapping);
    DescriptorOrdering new_order = results.get_descriptor_ordering();
    if (auto parsed_ordering = parsed_query.get_ordering()) {
        new_order.append(*parsed_ordering);
    }
    return Results(m_realm, results.get_query().and_query(std::move(parsed_query)), new_order);
```
I'm adding the append in https://github.com/realm/realm-core/pull/4291

Reply:
Thank you very much. I've modified as you suggested and I'll wait for the PR 4291 to be merged, to try test the code.

Reply:
You can temporarily point this PR to James's branch to get things building.

Reply:
As expected it works. Thank you.